internal/olm/installer/namespace_patcher.go

```diff
@@ -0,0 +1,106 @@
+package installer
+
+import (
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"strings"
+)
+
+type namespacePatcher struct {
+	uc runtime.UnstructuredConverter
+}
+
+func newNamespacePatcher() *namespacePatcher {
+	return &namespacePatcher{
+		uc: runtime.DefaultUnstructuredConverter,
+	}
+}
+
+func (np *namespacePatcher) setObjectsNamespace(objs []unstructured.Unstructured, namespace string) ([]unstructured.Unstructured, error) {
+	result := make([]unstructured.Unstructured, 0, len(objs))
+	for i := range objs {
+		obj := &objs[i]
+		if err := np.setNamespace(obj, namespace); err != nil {
+			return nil, err
+		}
+
+		result = append(result, *obj)
+	}
+
+	return result, nil
+}
+
+func (np *namespacePatcher) setNamespace(obj *unstructured.Unstructured, namespace string) error {
+	objKind := obj.GetObjectKind().GroupVersionKind().Kind
+	if obj.GetNamespace() == DefaultOLMNamespace {
+		obj.SetNamespace(namespace)
+	}
+
+	if objKind == "Namespace" && obj.GetName() == DefaultOLMNamespace {
+		obj.SetName(namespace)
+	}
+
+	if objKind == "ClusterRoleBinding" {
+		err := np.setClusterRoleBindingNamespace(obj, namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	if objKind == "Deployment" && obj.GetName() == catalogOperatorName {
+		err := np.setDeploymentArgNamespace(obj, namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (np *namespacePatcher) setDeploymentArgNamespace(obj *unstructured.Unstructured, namespace string) error {
+	var deploy appsv1.Deployment
+	if err := np.uc.FromUnstructured(obj.Object, &deploy); err != nil {
+		return err
+	}
+	args := deploy.Spec.Template.Spec.Containers[0].Args
+	for i, arg := range args {
+		if arg == "-namespace" || arg == "--namespace" {
+			args[i+1] = namespace
```

Comment:
This overwrites the next argument.

Reply:
The reason why we overwrite the next argument is because the current one is equal to `-namespace` and we expect the actual value to come as a following argument. If the problem is the potential overflow, I'd be happy to add a guard.